src/Lexers/ModelLexer.php

```diff
@@ -152,6 +157,10 @@ private function buildModel(string $name, array $columns)
             unset($columns['softdeletestz']);
         }
 
+        if ($model->usesSoftDeletes()) {
```

Comment:
The method we used before checked if `usesSoftDeletes` returns true and adds the `softDeletes()` at the end of the migration.

Reply:
Again, seems like "new" code. I appreciate this one has some tests added for it. Ultimately I feel this PR should only change the `MigrationGenerator`. It seems like it could cause other bugs to suddenly add a a column to models.

Reply:
It’s because we’re adding the column manually, instead of just setting a flag, in the correct order it was defined that a few refactoring was required in other places but no previous functionally has changed.